src/lib/autoDetect.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  ExtensionContext,
+  workspace,
+  window,
+  ConfigurationTarget,
+} from 'vscode';
+import { FileAssociations } from './types';
+const { extname } = require('path');
+import {
+  EXTENSION_CONFIG_NAME,
+  EXTENSION_CONFIG_KEYS,
+  HUBL_HTML_ID,
+  HUBL_CSS_ID,
+} from './constants';
+
+export const initializeHubLAutoDetect = (context: ExtensionContext) => {
+  if (
+    hasHubLAssociations() ||
+    checkGlobalState() ||
```

Comment:
nit: `checkGlobalState` feels a little vague, I like `optedOutGlobally`. Maybe just my weird brain but I read code aloud in my head and "if has HubL associations or check global state" flows a little worse than "if has HubL associations or opted out globally"